### PR TITLE
Add pagination to model query

### PR DESCRIPTION
**Pull request by fullonic**
This adds the pagination capabilities to the model query. There is already an issue #52  questioning about how to do pagination with orm.
This implementation is influenced by the Flask-SQLAlchemy pagination method. I just removed all the stuff related with http and adapted it to fit with orm.

It's a draft because it doesn't provided any documentation example (maybe some comments and docstrings?) and I still have some doubts. See comments please.
There is also provided some tests on `tests/test_models.py::test_pagination` where we can see some use cases

### CODE REVIEW

File: orm/models.py [hunk 1]
```diff
@@ -403,7 +403,7 @@ async def prev(self):
         return await self.query.paginate(self.page - 1, self.per_page)
 
     async def iterate(self):
-        yield await self.prev()
+        yield self
```
Comment (fullonic) [inline orm/models.py:406]: When we create the pagination query, it already returns the first page requested. Therefore if we want to iterate through all pages we must return again the first page, and then start looping over the remaining pages.

File: orm/models.py [hunk 2]
```diff
@@ -266,6 +266,14 @@ async def create(self, **kwargs):
         instance.pk = await self.database.execute(expr)
         return instance
 
+    async def paginate(self, page:int = 1, per_page: int = 10):
```
Comment (fullonic) [inline orm/models.py:270]: It's fine we provide default values?
Also, what is the best approach if the user pass a zero or negative number to page and per_page args. Do some if/ else checks and raise a exception or delegate to the user the responsibility to ensure that values are non zero positive integers?

File: orm/models.py [hunk 3]
```diff
@@ -396,14 +397,10 @@ def next_num(self):
         return self.page + 1
 
     async def next(self):
-        if self.has_next:
-            return await self.query.paginate(self.page + 1, self.per_page)
-        return await self.query.paginate(self.page, self.per_page)
+        return await self.query.paginate(self.page + 1, self.per_page)
```
Comment (fullonic) [inline orm/models.py:400]: I have doubts about want would be the best to return when there are no more next pages.
Here would be return a empty when an user request next and there are no next page.
It's fine like this, or should we return the last page again? or raise a warning?

**Comment by euri10** [comment]
I have doubts this should be implemented the offset way.
https://use-the-index-luke.com/no-offset

**Comment by tomchristie** [comment]
Thanks @fullonic - there's *absolutely* some good stuff here, but I don't really want pagination baked directly into the ORM package itself. There's a bunch of different implementations approaches we could take, and figuring out a generic pagination API would be a bunch of work that I don't have time to think about or review at the moment. At least at the moment something like this would be better off as a third party package, or as a gist that we can link users to as an example.

**Comment by fullonic** [comment]
@tomchristie I do understand. Also after reading the post suggested by @euri10, the implementation here isn't the best either. 
I also apology for making the PR before discuss it first on the chat.

**Pull request closed without merging**